### ISSUE TITLE
Endpoint current_weather does not work properly?

### ISSUE THREAD

**Issue by tawissus** [jdemaeyer/brightsky#168]
Heyho,

In Brandenburg, for example, I only get NULL-Values:
https://api.brightsky.dev/current_weather?lat=53.077892&lon=13.228488

`{"weather":{"source_id":144356,"timestamp":"2024-10-12T13:30:00+00:00","cloud_cover":null,"condition":null,"dew_point":3.95,"solar_10":null,"solar_30":null,"solar_60":null,"precipitation_10":0.0,"precipitation_30":0.0,"precipitation_60":0.0,"pressure_msl":null,"relative_humidity":57,"visibility":null,"wind_direction_10":null,"wind_direction_30":null,"wind_direction_60":null,"wind_speed_10":null,"wind_speed_30":null,"wind_speed_60":null,"wind_gust_direction_10":null,"wind_gust_direction_30":null,"wind_gust_direction_60":null,"wind_gust_speed_10":null,"wind_gust_speed_30":null,"wind_gust_speed_60":null,"sunshine_30":null,"sunshine_60":null,"temperature":12.2,"icon":"clear-day"},"sources":[{"id":144356,"dwd_station_id":"03509","observation_type":"synop","lat":53.102,"lon":13.0421,"height":78.0,"station_name":"Stechlin-Menz","wmo_station_id":"F151","first_record":"2024-10-11T07:30:00+00:00","last_record":"2024-10-12T13:30:00+00:00","distance":12747.0}]}`

Berlin and Hamburg work well in comparison.
https://api.brightsky.dev/current_weather?lat=52.486312&lon=13.416678
https://api.brightsky.dev/current_weather?lat=53.518099&lon=9.922151

What could be going wrong?

**Comment by jdemaeyer** [comment]
Hi @tawissus, thanks for the report!

Judging from the [nearest sources to your coordinates](https://api.brightsky.dev/sources?lat=53.077892&lon=13.228488), it seems that that area is full of either defunct or badly equipped stations:

* [station 03509 at 13 km](https://api.brightsky.dev/current_weather?dwd_station_id=03509)
* [station 05745 at 14 km](https://api.brightsky.dev/current_weather?dwd_station_id=05745)
* [station S360 at 14 km](https://api.brightsky.dev/current_weather?wmo_station_id=S360)
* [station S372 at 19 km](https://api.brightsky.dev/current_weather?wmo_station_id=S372)
* [station S350 at 19 km](https://api.brightsky.dev/current_weather?wmo_station_id=S350)
* [station S377 at 23 km](https://api.brightsky.dev/current_weather?wmo_station_id=S377)
* [station 01810 at 28 km](https://api.brightsky.dev/current_weather?dwd_station_id=01810)

Finally, [station 00096](https://api.brightsky.dev/current_weather?dwd_station_id=00096) seems to be the nearest station with a somewhat full set of parameters.

Our algorithm that fills up incomplete data is quite straightforward (aka stupid) to avoid running too many queries against the database, essentially [trying to find a close station with _all_ missing parameters](https://github.com/jdemaeyer/brightsky/blob/master/brightsky/query.py#L145-L147). In this case, such a station doesn't exist, as 03509 misses almost everything and even 00096 does not measure sunshine or solar irradiance.

Long story short (writing all that down just helped me to get my head straight ;)), we've had similar issues for the `/weather` endpoint before (which uses a similar but slightly more elaborate fill-up mechanism). I have been wanting to touch the fill-up mechanism for a while now and make it more aggressive, and this seems like a good reason to finally start. That'll take a while as I'm a little short on capacity right now, unfortunately. Let me check if I can get a rough hotfix out first, I'll update this tomorrow at the latest.

tl;dr: gimme a sec ;)

**Comment by jdemaeyer** [comment]
https://github.com/jdemaeyer/brightsky/commit/d31a834e7fb392e7726cc807d2134fb23782c0b5

Just released a small band-aid for this issue, Bright Sky should now at least include the data from station 00096 for your coordinates!

**Comment by tawissus** [comment]
Hey @jdemaeyer,
Thanks. You are great! I'm getting data again. I didn't think the weather stations were so fragile.
But now I'm wondering what the state of the weather stations is like. Are there so many different models or have many of them broken over time? But only the DWD knows that. Actually, they should put you on a roll of honor. 😄
